File: app/src/main/java/com/zacharee1/systemuituner/misc/CustomAdapter.java
package com.zacharee1.systemuituner.misc;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.zacharee1.systemuituner.R;
import com.zacharee1.systemuituner.activites.AppsListActivity;
import com.zacharee1.systemuituner.activites.ComponentsListActivity;
import com.zacharee1.systemuituner.fragmenthelpers.LockHelper;

import java.util.ArrayList;

public class CustomAdapter extends RecyclerView.Adapter<CustomAdapter.CustomHolder> {
    private ArrayList<AppInfo> apps = new ArrayList<>();
    private Context context;
    private boolean isLeft;

    public CustomAdapter(ArrayList<AppInfo> appInfos, Context context, boolean isLeft) {
        apps = appInfos;
        this.context = context;
        this.isLeft = isLeft;
    }

    @Override
    public int getItemCount() {
        return apps.size();
    }

    @Override
    public CustomHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.app_info_layout, parent, false);
        return new CustomHolder(view);
    }

    @Override
    public void onBindViewHolder(CustomHolder holder, int position) {
        holder.setAppInfo(apps.get(position));
    }

    public class CustomHolder extends RecyclerView.ViewHolder {
        public AppInfo appInfo;
        public View view;

        CustomHolder(View v) {
            super(v);
            view = v;
        }

        public void setAppInfo(AppInfo info) {
            appInfo = info;

            setAppName();
            setAppIcon();
            setListener();
        }

        public void setAppName() {
            TextView name = view.findViewById(R.id.app_name);
            name.setText(appInfo.appName);
        }

        public void setAppIcon() {
            ImageView icon = view.findViewById(R.id.app_icon);
            icon.setImageDrawable(appInfo.appIcon);
        }

        private void setListener() {
            view.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (context instanceof AppsListActivity) {
                        Intent activity = new Intent(context, ComponentsListActivity.class);
                        activity.putExtra("package", appInfo.packageName);
                        activity.putExtra("name", appInfo.appName);
                        activity.putExtra("isLeft", isLeft);

                        ((Activity) context).startActivityForResult(activity, 1337);
                    } else if (context instanceof ComponentsListActivity) {
                        SettingsUtils.writeSecure(context, isLeft ? "sysui_keyguard_left" : "sysui_keyguard_right", appInfo.packageName + "/" + appInfo.componentName);
                        Activity activity = (Activity) context;
                        activity.setResult(Activity.RESULT_OK);
                        activity.finish();
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/zacharee1/systemuituner/misc/CustomAdapter.kt
package com.zacharee1.systemuituner.misc

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.rey.material.widget.CheckedImageView
import com.zacharee1.systemuituner.R
import com.zacharee1.systemuituner.activites.apppickers.AppsListActivity
import com.zacharee1.systemuituner.activites.apppickers.ComponentsListActivity
import com.zacharee1.systemuituner.util.writeSecure
import java.util.*

class CustomAdapter(private val apps: ArrayList<AppInfo>,
                    private val context: Context,
                    private val isLeft: Boolean,
                    private val singleSelect: Boolean) : RecyclerView.Adapter<CustomAdapter.CustomHolder>() {

    override fun getItemCount(): Int {
        return apps.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomHolder {
        val view = LayoutInflater.from(context).inflate(if (singleSelect) R.layout.app_info_single else R.layout.app_info_multi, parent, false)
        return CustomHolder(view)
    }

    override fun onBindViewHolder(holder: CustomHolder, position: Int) {
        holder.setAppInfo(apps[position])
    }

    inner class CustomHolder internal constructor(var view: View) : RecyclerView.ViewHolder(view) {
        private lateinit var appInfo: AppInfo

        fun setAppInfo(info: AppInfo) {
            appInfo = info

            setAppName()
            setAppIcon()
            setAppSummary()
            setListener()
        }

        private fun setAppName() {
            val name = view.findViewById<TextView>(R.id.app_name)
            name.text = appInfo.appName
        }

        private fun setAppIcon() {
            val icon = view.findViewById<ImageView>(R.id.app_icon)
            icon.setImageDrawable(appInfo.appIcon)
        }

        private fun setAppSummary() {
            val summary = view.findViewById<TextView>(R.id.summary)
            summary.text = appInfo.componentName
        }

        private fun setListener() {
            view.setOnClickListener {
                val check = view.findViewById<CheckedImageView>(R.id.checkbox)
                check.isSelected = true

                if (context is AppsListActivity) {
                    val activity = Intent(context, ComponentsListActivity::class.java)
                    activity.putExtra("package", appInfo.packageName)
                    activity.putExtra("name", appInfo.appName)
                    activity.putExtra("isLeft", isLeft)

                    (context as Activity).startActivityForResult(activity, 1337)
                } else if (context is ComponentsListActivity) {
                    context.writeSecure(
                            if (isLeft) "sysui_keyguard_left" else "sysui_keyguard_right",
                            appInfo.packageName + "/" + appInfo.componentName
                    )
                    val activity = context as Activity
                    activity.setResult(Activity.RESULT_OK)
                    activity.finish()
                }
            }
        }
    }
}
